test/mocks/observableApi.ts

```diff
@@ -342,25 +346,46 @@ const workspaces = [
     name: "Mock User's Workspace",
     tier: "pro_2024",
     type: "team",
-    role: "member"
+    role: "member",
+    projects_info: []
   },
   {
     id: "0000000000000002",
     login: "mock-user-ws-2",
     name: "Mock User Second Workspace",
     tier: "pro_2024",
     type: "team",
-    role: "owner"
+    role: "owner",
+    projects_info: []
   },
   {
     id: "0000000000000003",
     login: "mock-user-ws-3",
     name: "Mock User's Third Workspace Wrong Tier",
     tier: "pro_2024",
     type: "team",
-    role: "viewer"
+    role: "viewer",
+    projects_info: []
+  },
+  {
+    id: "0000000000000004",
+    login: "mock-user-ws-4",
+    name: "Mock User's Fourth Workspace Guest Member as Editor",
+    tier: "pro_2024",
+    type: "team",
+    role: "guest_member",
+    projects_info: [{project_slug: "test-project-1", project_role: "editor"}]
+  },
+  {
+    id: "0000000000000005",
+    login: "mock-user-ws-5",
+    name: "Mock User's Fifth Workspace Guest Member as Viewer",
+    tier: "pro_2024",
+    type: "team",
+    role: "guest_member",
+    projects_info: [{project_slug: "test-project-2", project_role: "viewer"}]
   }
-];
+] as GetCurrentUserResponse["workspaces"];
```

Comment:
`as` will mask any errors here, which can cause problems down the line. If you are trying to mask an error, please make this actually match the type so the error goes away. If you are trying to make sure that TS sees this as the right kind of value, some approaches that are helpful are:

1. type the value: `const workspaces: GetcurrentUserResponse["workspaces"] = [...]`
2. replace the `as` with `satisfies` which does a safe type check without masking errors.
3. use `as const` which often helps TS realize the right type.
3.